### PR TITLE
Imported M1-related changes from

### PR DESCRIPTION
**Pull request by GaroRobe**
1. Imported M1 support fixes from https://github.com/kamilsa/soramitsu-hunter
2. Added support for both x86_64 and arm64 MacOS
3. Added new OpenSSL versions, up to 1.1.1l

### CODE REVIEW

File: cmake/projects/OpenSSL/hunter.cmake [hunk 1]
```diff
@@ -65,6 +65,95 @@ hunter_add_version(
...
+    "https://github.com/openssl/openssl/archive/OpenSSL_1_1_1k.tar.gz"
+    SHA1
+    95a5b6a0f9e866b1b131d71ec13b7f2b83a93da1
+)
+
+hunter_add_version(
+    PACKAGE_NAME
+    OpenSSL
+    VERSION
+    "1.1.1k"
```
Comment (xDimon) [inline cmake/projects/OpenSSL/hunter.cmake:150]: typo?

File: cmake/projects/OpenSSL/hunter.cmake [hunk 2]
```diff
@@ -65,6 +65,95 @@ hunter_add_version(
...
+hunter_add_version(
+    PACKAGE_NAME
+    OpenSSL
+    VERSION
+    "1.1.1j"
+    URL
+    "https://github.com/openssl/openssl/archive/OpenSSL_1_1_1j.tar.gz"
+    SHA1
+    327a400b8bb58058604f83c7e839fea11bba095e
+)
```
Comment (xDimon) [inline cmake/projects/OpenSSL/hunter.cmake:133]: Duplicate

File: cmake/projects/OpenSSL/hunter.cmake [hunk 3]
```diff
@@ -65,6 +65,95 @@ hunter_add_version(
...
+    OpenSSL
+    VERSION
+    "1.1.1k"
+    URL
+    "https://github.com/openssl/openssl/archive/OpenSSL_1_1_1k.tar.gz"
+    SHA1
+    95a5b6a0f9e866b1b131d71ec13b7f2b83a93da1
+)
+
+hunter_add_version(
```
Comment (xDimon) [inline cmake/projects/OpenSSL/hunter.cmake:146]: Perhaps, it is better to leave really used version

File: cmake/projects/OpenSSL/hunter.cmake (hunk 1 above)
Comment (GaroRobe) [reply cmake/projects/OpenSSL/hunter.cmake:150]: yep

File: cmake/projects/OpenSSL/hunter.cmake (hunk 3 above)
Comment (GaroRobe) [reply cmake/projects/OpenSSL/hunter.cmake:146]: Wasn't OpenSSL update one of our decisions?

Comment (xDimon) [reply cmake/projects/OpenSSL/hunter.cmake:135]: It about configs of others versions.

**Review by xDimon — APPROVED** [review]
Looks good. Need to smoke test over build some project using new openssl version over this version of hunter

**Pull request merged**